models/ed/R/model2netcdf.ED2.R

```diff
@@ -103,18 +103,18 @@ model2netcdf.ED2 <- function(outdir, sitelat, sitelon, start_date, end_date, pft
                                outdir, start_date, end_date, 
                                pft_names)
     }
-    
+    #length(out_list[["-T-"]]); length(out_list[["-E-"]])
     
     if (y == strftime(start_date, "%Y")) {
-      begins <- as.numeric(strftime(start_date, "%j")) - 1
+      begin_date <- as.Date(strftime(start_date))
     } else {
-      begins <- 0
+      begin_date <- base::as.Date(paste0(y,"-01-01"))
     }
     
     if (y == strftime(end_date, "%Y")) {
       ends <- as.numeric(strftime(end_date, "%j"))
     } else {
-      ends <- as.numeric(strftime(paste0(y, "-12-31"), "%j")) 
+      ends <- as.numeric(strftime(paste0(y, "-12-31"), "%j"))
     }
```

Comment:
you mean into an ifelse(,,,) instead of if () {} else {}?

Reply:
No. I avoid `ifelse` like the plague because of its implicit coercion rules. `dplyr::if_else` is better because it does aggressive type checking and fails loudly if anything is out of order.

But I actually meant this:

```
if (y == strftime(end_date, "%Y")) {
  begin_date <- ...
  ends <- ...
} else {
  begin_date <- ...
  ends <- ...
}
```

Reply:
Actually, forget I said that -- I just saw that it's `begin_date` and `end_date`, so they are different logic. Sorry!